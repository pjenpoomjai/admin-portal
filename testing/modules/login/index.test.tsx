import { act, fireEvent, render, waitFor } from '@testing-library/react'
import CommonDialogContext from 'contexts/commonDialog/Context'
import LoginContainer from 'modules/login'
import { useSearchParams } from 'next/navigation'
import CommonDialogProvider from 'providers/CommonDialogProvider'
import {
	buildMockUseAppContext,
	mockHandleRedirectPromise,
} from 'testing/test-utils/buildMockUseAppContext'
import { useAppContext } from 'x-core-modules/builder/appBuilder'

jest.mock('next/navigation', () => ({
	...jest.requireActual('next/navigation'),
	useSearchParams: jest.fn(),
}))
jest.mock('x-core-modules/builder/appBuilder', () => ({
	...jest.requireActual('x-core-modules/builder/appBuilder'),
	useAppContext: jest.fn(),
}))
jest.mock('config/msal', () => ({
	msalConfig: {
		auth: {
			redirectUri: '',
		},
	},
}))

const mockUseAppContext = useAppContext as jest.Mock
const mockUseSearchParams = useSearchParams as jest.Mock

describe('LoginContainer : Container Configuration', () => {
	beforeEach(() => {
		mockUseSearchParams.mockReturnValue({
			get: jest.fn().mockReturnValue(undefined),
		})
	})

	afterEach(() => {
		jest.resetModules()
		jest.resetAllMocks()
	})

	test('Should render successfully', () => {
		mockUseAppContext.mockReturnValue(
			buildMockUseAppContext({
				overrideAdapterQuery: {
					lazyQuery: jest.fn().mockImplementation(() => {
						return { data: undefined, error: undefined, trigger: jest.fn() }
					}),
				},
			}),
		)
		const { baseElement } = render(<LoginContainer />)
		expect(baseElement).toMatchSnapshot()
		expect(mockHandleRedirectPromise).toHaveBeenCalled()
	})

	test('Should display error dialog if authentication failed', async () => {
		// mock state
		const token = '0e798be3-45b4-45a8-8042-efa6c5407286'
		const expiresOn = 1706866713618
		const authenticationResult = {
			accessToken: token,
			expiresOn,
		}
		mockUseAppContext.mockReturnValue(
			buildMockUseAppContext({
				msalProviderSelector: {
					useAuthenticationResultSelector: () => authenticationResult,
				},
				commonDialogContextSelector: CommonDialogContext.selector,
				overrideAdapterQuery: {
					lazyQuery: jest
						.fn()
						.mockImplementationOnce(({ option }) => {
							option.onError(new Error())
							return { data: undefined, error: undefined, trigger: jest.fn() }
						})
						.mockImplementation(() => {
							return { data: undefined, error: undefined, trigger: jest.fn() }
						}),
					mutation: jest.fn().mockImplementation(() => {
						return { trigger: jest.fn() }
					}),
				},
			}),
		)
		// render
		const { getByTestId, queryByTestId, baseElement } = render(
			<CommonDialogContext.Provider>
				<CommonDialogProvider.Provider>
					<LoginContainer />
				</CommonDialogProvider.Provider>
			</CommonDialogContext.Provider>,
		)
		// expect
		await waitFor(() => {
			expect(getByTestId('no-permission-title-dialog')).toBeInTheDocument()
		})
		expect(baseElement).toMatchSnapshot()

		const closeDialogButton = getByTestId('no-permission-primary-button-button')
		expect(closeDialogButton).toBeInTheDocument()
		act(() => {
			fireEvent.click(closeDialogButton)
		})

		await waitFor(() => {
			expect(queryByTestId('no-permission-title-dialog')).toBeNull()
		})
		expect(baseElement).toMatchSnapshot()
		expect(mockHandleRedirectPromise).not.toHaveBeenCalled()
	})
})
