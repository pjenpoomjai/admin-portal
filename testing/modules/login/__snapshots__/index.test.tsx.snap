// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginContainer : Container Configuration Should display error dialog if authentication failed 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.75;
  letter-spacing: 0.02857em;
  text-transform: uppercase;
  min-width: 64px;
  padding: 6px 16px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  color: #fff;
  background-color: #1976d2;
  box-shadow: 0px 3px 1px -2px rgba(0,0,0,0.2),0px 2px 2px 0px rgba(0,0,0,0.14),0px 1px 5px 0px rgba(0,0,0,0.12);
  width: 100%;
  box-shadow: unset;
}

.emotion-0::-moz-focus-inner {
  border-style: none;
}

.emotion-0.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-0 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-0:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #1565c0;
  box-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);
}

@media (hover: none) {
  .emotion-0:hover {
    background-color: #1976d2;
  }
}

.emotion-0:active {
  box-shadow: 0px 5px 5px -3px rgba(0,0,0,0.2),0px 8px 10px 1px rgba(0,0,0,0.14),0px 3px 14px 2px rgba(0,0,0,0.12);
}

.emotion-0.Mui-focusVisible {
  box-shadow: 0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12);
}

.emotion-0.Mui-disabled {
  color: rgba(0, 0, 0, 0.26);
  box-shadow: none;
  background-color: rgba(0, 0, 0, 0.12);
}

.emotion-0.MuiButton-root {
  text-transform: none;
  color: var(--gray-30);
  border-radius: 6px;
  background-color: var(--violet-500);
}

.emotion-0:hover {
  background-color: var(--violet-700);
}

.emotion-0:disabled {
  background-color: var(--gray-300);
}

.emotion-0:focus {
  border: 3px solid var(--violet-300);
}

.emotion-1 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

.emotion-2 {
  position: fixed;
  z-index: 1300;
  right: 0;
  bottom: 0;
  top: 0;
  left: 0;
}

@media print {
  .emotion-2 {
    position: absolute!important;
  }
}

.emotion-3 {
  position: fixed;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  right: 0;
  bottom: 0;
  top: 0;
  left: 0;
  background-color: rgba(0, 0, 0, 0.5);
  -webkit-tap-highlight-color: transparent;
  z-index: -1;
}

.emotion-4 {
  height: 100%;
  outline: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

@media print {
  .emotion-4 {
    height: auto;
  }
}

.emotion-5 {
  background-color: #fff;
  color: rgba(0, 0, 0, 0.87);
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 4px;
  box-shadow: 0px 11px 15px -7px rgba(0,0,0,0.2),0px 24px 38px 3px rgba(0,0,0,0.14),0px 9px 46px 8px rgba(0,0,0,0.12);
  margin: 32px;
  position: relative;
  overflow-y: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  max-height: calc(100% - 64px);
  max-width: 600px;
}

@media print {
  .emotion-5 {
    overflow-y: visible;
    box-shadow: none;
  }
}

@media (max-width:663.95px) {
  .emotion-5.MuiDialog-paperScrollBody {
    max-width: calc(100% - 64px);
  }
}

.emotion-6 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 500;
  font-size: 1.25rem;
  line-height: 1.6;
  letter-spacing: 0.0075em;
  padding: 16px 24px;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

.emotion-7 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  -webkit-overflow-scrolling: touch;
  overflow-y: auto;
  padding: 20px 24px;
}

.MuiDialogTitle-root+.emotion-7 {
  padding-top: 0;
}

.emotion-8 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
  color: rgba(0, 0, 0, 0.6);
}

.emotion-9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 8px;
  -webkit-box-pack: end;
  -ms-flex-pack: end;
  -webkit-justify-content: flex-end;
  justify-content: flex-end;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

.emotion-9>:not(style)~:not(style) {
  margin-left: 8px;
}

.emotion-10 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.75;
  letter-spacing: 0.02857em;
  text-transform: uppercase;
  min-width: 64px;
  padding: 6px 16px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  color: #fff;
  background-color: #1976d2;
  box-shadow: 0px 3px 1px -2px rgba(0,0,0,0.2),0px 2px 2px 0px rgba(0,0,0,0.14),0px 1px 5px 0px rgba(0,0,0,0.12);
  box-shadow: unset;
}

.emotion-10::-moz-focus-inner {
  border-style: none;
}

.emotion-10.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-10 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-10:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #1565c0;
  box-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);
}

@media (hover: none) {
  .emotion-10:hover {
    background-color: #1976d2;
  }
}

.emotion-10:active {
  box-shadow: 0px 5px 5px -3px rgba(0,0,0,0.2),0px 8px 10px 1px rgba(0,0,0,0.14),0px 3px 14px 2px rgba(0,0,0,0.12);
}

.emotion-10.Mui-focusVisible {
  box-shadow: 0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12);
}

.emotion-10.Mui-disabled {
  color: rgba(0, 0, 0, 0.26);
  box-shadow: none;
  background-color: rgba(0, 0, 0, 0.12);
}

.emotion-10.MuiButton-root {
  text-transform: none;
  color: var(--gray-30);
  border-radius: 6px;
  background-color: var(--violet-500);
}

.emotion-10:hover {
  background-color: var(--violet-700);
}

.emotion-10:disabled {
  background-color: var(--gray-300);
}

.emotion-10:focus {
  border: 3px solid var(--violet-300);
}

<body
  style="padding-right: 1024px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  >
    <div
      data-id="root-login-div"
      data-testid="root-login-div"
    >
      <div
        class="flex justify-center items-center h-full w-full min-h-screen absolute overflow-hidden top-0 left-0"
      >
        <svg
          class="w-full h-full absolute -z-10 bg-gray-900"
          type="mockSvg"
        />
        <div
          class="w-80 px-10 py-20 bg-gray-30 rounded shadow-md bg-black bg-opacity-7"
        >
          <div
            class="mb-12 flex flex-col items-center"
          >
            <div
              class="flex flex-row gap-3 items-center pr-6"
            >
              <svg
                type="mockSvg"
              />
              <p
                class="block antialiased font-sans font-light text-2xl text-violet-900"
                data-id="pymd-label-typography"
                data-testid="pymd-label-typography"
                placeholder=""
              >
                Admin portal
              </p>
            </div>
            <p
              class="block antialiased font-sans text-base font-light leading-relaxed text-inherit pt-2"
              data-id="pymd-description-typography"
              data-testid="pymd-description-typography"
              placeholder=""
            >
              Payment Domain Web Admin
            </p>
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-fullWidth text-sm font-semibold h-[34px] emotion-0"
            data-id="sign-in-button"
            data-testid="sign-in-button"
            id="sign-in"
            tabindex="0"
            type="button"
          >
            Sign in
            <span
              class="MuiTouchRipple-root emotion-1"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiDialog-root MuiModal-root emotion-2"
    data-id="no-permission-dialog"
    data-testid="no-permission-dialog"
    id="no-permission"
    role="presentation"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root MuiModal-backdrop emotion-3"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-testid="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiDialog-container MuiDialog-scrollPaper emotion-4"
      role="presentation"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      tabindex="-1"
    >
      <div
        aria-describedby="no-permission-content"
        aria-labelledby="no-permission-title"
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm emotion-5"
        role="dialog"
        style="min-width: 386px; min-height: 274px; border-radius: 16px;"
      >
        <div
          class="px-[24px] pt-[24px] py-[16px] flex flex-col break-all flex-auto overflow-auto"
        >
          <div
            class="self-center"
          >
            <svg
              class="m-2"
              type="mockSvg"
            />
          </div>
          <div
            class="self-center flex-auto w-full overflow-auto"
          >
            <h2
              class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root text-center text-lg text-gray-900 font-semibold px-0 pt-[24px] pb-[8px] emotion-6"
              data-id="no-permission-title-dialog"
              data-testid="no-permission-title-dialog"
              id="no-permission-title"
            >
              Something went wrong
            </h2>
            <div
              class="MuiDialogContent-root px-0 pb-[24px] w-full emotion-7"
            >
              <p
                class="MuiTypography-root MuiDialogContentText-root MuiTypography-body1 text-center text-base text-gray-500 break-words break-normal emotion-8"
                id="no-permission-content"
              >
                Please try again.
              </p>
            </div>
          </div>
          <div
            class="MuiDialogActions-root MuiDialogActions-spacing p-0 contents pt-1 emotion-9"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary !bg-indigo-500 mx-auto emotion-10"
              data-id="no-permission-primary-button-button"
              data-testid="no-permission-primary-button-button"
              id="primary-button"
              style="min-width: 125px;"
              tabindex="0"
              type="button"
            >
              OK
              <span
                class="MuiTouchRipple-root emotion-1"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      data-testid="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;

exports[`LoginContainer : Container Configuration Should display error dialog if authentication failed 2`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.75;
  letter-spacing: 0.02857em;
  text-transform: uppercase;
  min-width: 64px;
  padding: 6px 16px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  color: #fff;
  background-color: #1976d2;
  box-shadow: 0px 3px 1px -2px rgba(0,0,0,0.2),0px 2px 2px 0px rgba(0,0,0,0.14),0px 1px 5px 0px rgba(0,0,0,0.12);
  width: 100%;
  box-shadow: unset;
}

.emotion-0::-moz-focus-inner {
  border-style: none;
}

.emotion-0.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-0 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-0:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #1565c0;
  box-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);
}

@media (hover: none) {
  .emotion-0:hover {
    background-color: #1976d2;
  }
}

.emotion-0:active {
  box-shadow: 0px 5px 5px -3px rgba(0,0,0,0.2),0px 8px 10px 1px rgba(0,0,0,0.14),0px 3px 14px 2px rgba(0,0,0,0.12);
}

.emotion-0.Mui-focusVisible {
  box-shadow: 0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12);
}

.emotion-0.Mui-disabled {
  color: rgba(0, 0, 0, 0.26);
  box-shadow: none;
  background-color: rgba(0, 0, 0, 0.12);
}

.emotion-0.MuiButton-root {
  text-transform: none;
  color: var(--gray-30);
  border-radius: 6px;
  background-color: var(--violet-500);
}

.emotion-0:hover {
  background-color: var(--violet-700);
}

.emotion-0:disabled {
  background-color: var(--gray-300);
}

.emotion-0:focus {
  border: 3px solid var(--violet-300);
}

.emotion-1 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

<body
  style=""
>
  <div>
    <div
      data-id="root-login-div"
      data-testid="root-login-div"
    >
      <div
        class="flex justify-center items-center h-full w-full min-h-screen absolute overflow-hidden top-0 left-0"
      >
        <svg
          class="w-full h-full absolute -z-10 bg-gray-900"
          type="mockSvg"
        />
        <div
          class="w-80 px-10 py-20 bg-gray-30 rounded shadow-md bg-black bg-opacity-7"
        >
          <div
            class="mb-12 flex flex-col items-center"
          >
            <div
              class="flex flex-row gap-3 items-center pr-6"
            >
              <svg
                type="mockSvg"
              />
              <p
                class="block antialiased font-sans font-light text-2xl text-violet-900"
                data-id="pymd-label-typography"
                data-testid="pymd-label-typography"
                placeholder=""
              >
                Admin portal
              </p>
            </div>
            <p
              class="block antialiased font-sans text-base font-light leading-relaxed text-inherit pt-2"
              data-id="pymd-description-typography"
              data-testid="pymd-description-typography"
              placeholder=""
            >
              Payment Domain Web Admin
            </p>
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-fullWidth text-sm font-semibold h-[34px] emotion-0"
            data-id="sign-in-button"
            data-testid="sign-in-button"
            id="sign-in"
            tabindex="0"
            type="button"
          >
            Sign in
            <span
              class="MuiTouchRipple-root emotion-1"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`LoginContainer : Container Configuration Should render successfully 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.75;
  letter-spacing: 0.02857em;
  text-transform: uppercase;
  min-width: 64px;
  padding: 6px 16px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  color: #fff;
  background-color: #1976d2;
  box-shadow: 0px 3px 1px -2px rgba(0,0,0,0.2),0px 2px 2px 0px rgba(0,0,0,0.14),0px 1px 5px 0px rgba(0,0,0,0.12);
  width: 100%;
  box-shadow: unset;
}

.emotion-0::-moz-focus-inner {
  border-style: none;
}

.emotion-0.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-0 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-0:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #1565c0;
  box-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);
}

@media (hover: none) {
  .emotion-0:hover {
    background-color: #1976d2;
  }
}

.emotion-0:active {
  box-shadow: 0px 5px 5px -3px rgba(0,0,0,0.2),0px 8px 10px 1px rgba(0,0,0,0.14),0px 3px 14px 2px rgba(0,0,0,0.12);
}

.emotion-0.Mui-focusVisible {
  box-shadow: 0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12);
}

.emotion-0.Mui-disabled {
  color: rgba(0, 0, 0, 0.26);
  box-shadow: none;
  background-color: rgba(0, 0, 0, 0.12);
}

.emotion-0.MuiButton-root {
  text-transform: none;
  color: var(--gray-30);
  border-radius: 6px;
  background-color: var(--violet-500);
}

.emotion-0:hover {
  background-color: var(--violet-700);
}

.emotion-0:disabled {
  background-color: var(--gray-300);
}

.emotion-0:focus {
  border: 3px solid var(--violet-300);
}

.emotion-1 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

<body>
  <div>
    <div
      data-id="root-login-div"
      data-testid="root-login-div"
    >
      <div
        class="flex justify-center items-center h-full w-full min-h-screen absolute overflow-hidden top-0 left-0"
      >
        <svg
          class="w-full h-full absolute -z-10 bg-gray-900"
          type="mockSvg"
        />
        <div
          class="w-80 px-10 py-20 bg-gray-30 rounded shadow-md bg-black bg-opacity-7"
        >
          <div
            class="mb-12 flex flex-col items-center"
          >
            <div
              class="flex flex-row gap-3 items-center pr-6"
            >
              <svg
                type="mockSvg"
              />
              <p
                class="block antialiased font-sans font-light text-2xl text-violet-900"
                data-id="pymd-label-typography"
                data-testid="pymd-label-typography"
                placeholder=""
              >
                Admin portal
              </p>
            </div>
            <p
              class="block antialiased font-sans text-base font-light leading-relaxed text-inherit pt-2"
              data-id="pymd-description-typography"
              data-testid="pymd-description-typography"
              placeholder=""
            >
              Payment Domain Web Admin
            </p>
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-fullWidth text-sm font-semibold h-[34px] emotion-0"
            data-id="sign-in-button"
            data-testid="sign-in-button"
            id="sign-in"
            tabindex="0"
            type="button"
          >
            Sign in
            <span
              class="MuiTouchRipple-root emotion-1"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;
